src/librustc/hir/map/definitions.rs

```diff
@@ -537,22 +560,22 @@ impl DefPathData {
         }
     }
 
-    pub fn as_symbol(&self) -> Symbol {
```

Comment:
If the direction with keeping the span in `DefPathData` is pursued, this method can be kept for compatibility and used in places where full identifier is not needed. That will significantly reduce the churn.